@xen-orchestra/backups/_cleanVm.js

```diff
@@ -10,6 +10,24 @@ const { limitConcurrency } = require('limit-concurrency-decorator')
 const { Task } = require('./Task.js')
 const { Disposable } = require('promise-toolbox')
 
+// checking the size of a vhd directory is costly
+// 1 Http Query per 1000 blocks
+// we only check size of all the vhd are VhdFiles
+function shouldComputeVhdsSize(vhds) {
+  return vhds.every(vhd => vhd instanceof VhdFile)
+}
+
+async function computeVhdsSize(handler, vhdPaths) {
+  return await Disposable.use(Disposable.all(vhdPaths.map(vhdPath => openVhd(handler, vhdPath))), async vhds => {
+    if (!shouldComputeVhdsSize(vhds)) {
+      // don't lose time computing vhd size if some vhds have a non computable size
+      return
+    }
+    const sizes = await asyncMap(vhds, vhd => vhd.getSize())
+    return sum(sizes)
+  })
+}
```

Comment:
- you can use an arrow function if it's a single return
- you don't need to `return await`
- you don't need to use `Disposable.all`, `Disposable.use` accepts an array of disposables
- prefer simple `if (cond) statement` to `if (!cond) return; statement`

```suggestion
const computeVhdsSize = (handler, vhdPaths) => Disposable.use(vhdPaths.map(vhdPath => openVhd(handler, vhdPath)), async vhds => {
    if (shouldComputeVhdsSize(vhds)) {
      const sizes = await asyncMap(vhds, vhd => vhd.getSize())
      return sum(sizes)
    }
  })
}
```